File: 2.OOP/concepts/MediumExample/design_patterns/Structural/Adapter/Adapter.js
class Square{
    constructor(side) {
        this.side = side
    }
    getArea(){
        return this.side * this.side
    }
}
class Circle{
    constructor(radius) {
        this.radius = radius
    }
    getArea(){
        return Math.PI * this.radius * this.radius
    }
}
class ShapeAreaCalculator{
    constructor(shape) {
        this.shape = shape
    }
    getAreaInCm(){
        return this.shape.getArea()*2.54
    }
}
const square = new Square(5)
const circle = new Circle(3)
const squareAdapter = new ShapeAreaCalculator(square)
const circleAdapter = new ShapeAreaCalculator(circle)
console.log(`Area in cm : ${circleAdapter.getAreaInCm()}`);
console.log(`Area in cm : ${squareAdapter.getAreaInCm()}`);

File: 2.OOP/concepts/MediumExample/design_patterns/Structural/Adapter/Adapter.py
import math
class Square:
    def __init__(self,side):
        self.side =side
    def getArea(self):
        return self.side * self.side

class Circle:
    def __init__(self,radius):
        self.radius = radius
    def getArea(self):
        return math.pi * self.radius * self.radius

class ShapeAreaCalculator:
    def __init__(self,shape):
        self.shape = shape
    def getAreaInCm(self):
        return self.shape.getArea() * 2.54 # This factor is used to convert the area from inches (assumed unit) to centimeters.

square = Square(5)
circle = Circle(3)
square_adapter = ShapeAreaCalculator(square)
circle_adapter = ShapeAreaCalculator(circle)

print(f'Area of square in cm: {square_adapter.getAreaInCm()}')
print(f'Area of circle in cm: {circle_adapter.getAreaInCm()}')